### PR TITLE
[log4cplus] fix Windows Intel C++ build (probably others like MinGW, Clang, etc)

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **log4cplus/all**
[log4cplus] fix Windows Intel C++ build (probably others like MinGW, Clang, etc)

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/log4cplus/all/conanfile.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Log4cplusConan(ConanFile):
 
     def requirements(self):
         if self.options.with_iconv:
-            self.requires.add('libiconv/1.15')
+            self.requires.add('libiconv/1.16')
```
Comment (jgsogo) [inline recipes/log4cplus/all/conanfile.py:40]:
```suggestion
            self.requires('libiconv/1.16')
```

**Comment by jgsogo** [comment]
Please, open another. PR with the fix from my review....

File: recipes/log4cplus/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/log4cplus/all/conanfile.py:40]: is it a big deal? both works from my experience. if `self.requires.add` has some disadvantages, we might need to add a hook check
/cc @uilianries

Comment (jgsogo) [reply recipes/log4cplus/all/conanfile.py:40]: Yes, it is. `self.requires.add` is not documented and can change at any moment. We don't want people copy/pasting from a wrong recipe in CCI. Here it happened with `build_requires`: https://github.com/conan-io/conan/issues/6667

Comment (SSE4) [reply recipes/log4cplus/all/conanfile.py:40]: that's documented (at least I can find it in docs, e.g. [here](https://conan-docs.readthedocs.io/en/feature-cmake-toolchain-file/howtos/create_installer_packages.html))

Comment (jgsogo) [reply recipes/log4cplus/all/conanfile.py:40]: That `https://conan-docs.readthedocs.io/` link belongs to a 0.XX version of the documentation...

Comment (uilianries) [reply recipes/log4cplus/all/conanfile.py:40]: self.requires.add is outdated

**Pull request merged**